tutorials/src/cpu/detector/detector_ray_scan.cpp

```diff
@@ -56,14 +69,15 @@ int main() {
     // Index of the volume that the ray origin lies in
     detray::dindex start_index{0u};
 
-    // Number of rays in theta and phi
-    unsigned int theta_steps{100u};
-    unsigned int phi_steps{100u};
     // Origin of the rays
     const detray::tutorial::point3 origin{0.f, 0.f, 0.f};
     auto ray_generator =
         detray::uniform_track_generator<ray_t>(theta_steps, phi_steps, origin);
 
+    // Geometry context for the ray scan & ray counter
+    typename detector_t::geometry_context gctx{};
+    std::size_t ray_counter{0u};
```

Comment:
You can use ```detray::views::enumerate``` (same way as in ACTS) instead of a counter. You will need to add
```
#include "detray/utils/ranges.hpp"
```

Reply:
Will be interesting to see if the track generator composes correctly with the enumerate range :)